### ISSUE TITLE
"CoCreateInstanceEx: Access is denied"

### ISSUE THREAD

**Issue by sensubha** [titanium-as/TitaniumAS.Opc.Client#43]
While connecting remote OPC DA server using TitaniumAs Opc client, getting "CoCreateInstanceEx: Access is denied". How to handle this issue?

**Comment by mjosephabie** [comment]
I also have this problem. My Opc server are rovisys opc90server. However when I tried to another opc, it works normally.

**Comment by crxmaniac** [comment]
I am also struggling with the same issue. How do I configure COM-DCOM settings. Do I have to add any code to make the application available in DCOM Config under Component services?

**Comment by red-oker** [comment]
El problema se resuelve así:
1- Presione Windows + R.
2- Escriba domcnfg y presione Enter.
3- Seleccione: Raiz -> Servicios de componentes -> Equipos -> Mi PC -> Configuración DCOM.
4- Localice el DCOM que corresponda a su aplicación OPC.
5- Clic derecho y luego clic en propiedades.
6- Clic en la pestaña de seguridad.
7- En las secciones de "Permisos de inicio y activación" y "Permisos de acceso", seleccione personalizado.
8- Agregue el usuario con el que se ejecuta su aplicación en ambas secciones, asegúrese de darle todos los permisos y listo.
9- Aplique los cambios y listo.

**Comment by crxmaniac** [comment]
> @crxmaniac
> 
> El problema se resuelve así:
> 
> 1- Presione Windows + R.
> 
> 2- Escriba domcnfg y presione Enter.
> 
> 3- Seleccione: Raiz -> Servicios de componentes -> Equipos -> Mi PC -> Configuración DCOM.
> 
> 4- Localice el DCOM que corresponda a su aplicación OPC.
> 
> 5- Clic derecho y luego clic en propiedades.
> 
> 6- Clic en la pestaña de seguridad.
> 
> 7- En las secciones de "Permisos de inicio y activación" y "Permisos de acceso", seleccione personalizado.
> 
> 8- Agregue el usuario con el que se ejecuta su aplicación en ambas secciones, asegúrese de darle todos los permisos y listo.
> 
> 9- Aplique los cambios y listo.
> 
> 
Great thanks
Will i have to do this for all computers. Client and server?  

Vegard

**Comment by red-oker** [comment]
> 
> 
> > @crxmaniac
> > El problema se resuelve así:
> > 1- Presione Windows + R.
> > 2- Escriba domcnfg y presione Enter.
> > 3- Seleccione: Raiz -> Servicios de componentes -> Equipos -> Mi PC -> Configuración DCOM.
> > 4- Localice el DCOM que corresponda a su aplicación OPC.
> > 5- Clic derecho y luego clic en propiedades.
> > 6- Clic en la pestaña de seguridad.
> > 7- En las secciones de "Permisos de inicio y activación" y "Permisos de acceso", seleccione personalizado.
> > 8- Agregue el usuario con el que se ejecuta su aplicación en ambas secciones, asegúrese de darle todos los permisos y listo.
> > 9- Aplique los cambios y listo.
> 
> Great thanks
> Will i have to do this for all computers. Client and server?
> 
> Vegard

No, solo para los servidores OPC. Otra posible solución es que tu aplicación cliente se ejecute con el mismo usuario que tu servidor OPC, según la documentación de windows, el problema es que las aplicaciones COM (tu cliente y el servidor OPC) están corriendo en usuarios distintos.
